01.multithreading/MultiThreading.Task5.Threads.SharedCollection/Program.cs

```diff
@@ -0,0 +1,63 @@
+﻿/*
+ * 5. Write a program which creates two threads and a shared collection:
+ * the first one should add 10 elements into the collection and the second should print all elements
+ * in the collection after each adding.
+ * Use Thread, ThreadPool or Task classes for thread creation and any kind of synchronization constructions.
+ */
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MultiThreading.Task5.Threads.SharedCollection
+{
+    class Program
+    {
+        private static readonly ConcurrentQueue<int> _sharedCollection = new ConcurrentQueue<int>();
+        private static readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1, 1);
+
+        static void Main(string[] args)
+        {
+            Console.WriteLine("5. Write a program which creates two threads and a shared collection:");
+            Console.WriteLine("the first one should add 10 elements into the collection and the second should print all elements in the collection after each adding.");
+            Console.WriteLine("Use Thread, ThreadPool or Task classes for thread creation and any kind of synchronization constructions.");
+            Console.WriteLine();
+
+            var addingTask = AddingTask();
+            var printingTask = PrintingTask();
```

Comment:
Your code didn't output the last collection with 10 elements. They even are not synchronized and print task can output same collection twice.

Reply:
Please, look at the difference between Semaphore and SemaphoreSlim. (just replace your semaphore type and run again)

Reply:
Fixed